src/mesh/replicated_mesh.C

```diff
@@ -425,6 +425,10 @@ void ReplicatedMesh::delete_elem(Elem * e)
 void ReplicatedMesh::renumber_elem(const dof_id_type old_id,
                                    const dof_id_type new_id)
 {
+  // This could be a no-op
+  if (old_id == new_id)
+    return;
+
```

Comment:
This (and the same update in distributed_mesh.C) seem like good optimizations to have independent of the rest of the PR, I wouldn't be opposed to merging them in a separate branch but also would understand if you don't feel like doing that. I assume they are already in a separate commit here, so that's probably fine as well.